backend/routes/project.js

```diff
@@ -297,6 +297,71 @@ router.post("/proposal/list", function (req, res) {
   );
 });
 
+router.get("/state/:projectId", async (req, res) => {
+  const levels = (completed) => {
+    // NOTE: Reference: https://github.com/oceanprotocol/oceandao/wiki#r12-update-funding-tiers
+    if (completed === 0) return { level: "New Project", ceiling: 3000 };
+    if (completed === 1) return { level: "Existing Project", ceiling: 10000 };
+    if (completed >= 2 && completed < 5)
+      return { level: "Experienced Project", ceiling: 20000 };
+    if (completed >= 5) return { level: "Veteran Project", ceiling: 20000 };
+  };
+
+  const projectId = req.params.projectId;
+  Proposal.find(
+    { projectId: projectId },
+    "proposalFundingRequested proposalTitle round proposalEarmark airtableRecordId"
+  )
+    .sort({ round: -1 })
+    .exec((err, proposals) => {
+      Project.findById(projectId, async (err, project) => {
+        if (err) {
+          res.status(400).send(err);
+        }
+        const airtableInfos = await getProposalRedisMultiple(
+          proposals.map((x) => x.airtableRecordId),
+          "."
+        );
+
+        const grantsProposed = proposals.length;
+        const grantsReceived = airtableInfos.filter(
+          (x) =>
+            x["Proposal State"] === "Granted" ||
+            x["Proposal State"] === "Funded"
+        ).length;
+        const grantsCompleted = proposals.filter(
+          (x) => x.delivered.status == 2
+        ).length;
+
+        const projectCategory = project.projectCategory;
+
+        const level = levels(grantsCompleted);
```

Comment:
wouldn't it be more feasible to just get all the calculated values from above from the project summary table? I guess those are not cached yet but if we cache them we could reduce a lot of code.